### ISSUE TITLE
[Package Issue]: Nilesoft.Shell

### ISSUE THREAD

**Issue by pagunduu** [microsoft/winget-pkgs#195989]
### Please confirm these before moving forward

- [x] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [x] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

Installation of Nilesoft.Shell Fails

### Steps to reproduce

1. Open up the terminal.
2. Try to install Nilesoft shell, `winget install Nilesoft.Shell`.
3. It fails.

### Actual behavior

Installation Fails.

### Expected behavior

The installer should have elevated permissions to successfully install.

### Environment

```raw
Windows Package Manager v1.9.25200
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.26100.2314
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.24.25199.0

Winget Directories
------------------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputD…
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.js…
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages
Installer Downloads                %USERPROFILE%\Downloads

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Disabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
ProxyCommandLineOptions                   Disabled
DefaultProxy                              Disabled
```

### Screenshots and Logs

```
=== Logging started: 03-12-2024  22:19:48 ===
Action start 22:19:48: INSTALL.
Action start 22:19:48: FindRelatedProducts.
Action ended 22:19:48: FindRelatedProducts. Return value 1.
Action start 22:19:48: OnUpdate.
Action ended 22:19:48: OnUpdate. Return value 1.
Action start 22:19:48: ValidateProductID.
Action ended 22:19:48: ValidateProductID. Return value 1.
Action start 22:19:48: CostInitialize.
Action ended 22:19:48: CostInitialize. Return value 1.
Action start 22:19:48: FileCost.
Action ended 22:19:48: FileCost. Return value 1.
Action start 22:19:48: CostFinalize.
Action ended 22:19:48: CostFinalize. Return value 1.
Action start 22:19:48: InstallValidate.
Action ended 22:19:48: InstallValidate. Return value 1.
Action start 22:19:48: InstallInitialize.
Action ended 22:19:48: InstallInitialize. Return value 1.
Action start 22:19:48: ProcessComponents.
Action ended 22:19:48: ProcessComponents. Return value 1.
Action start 22:19:48: UnpublishFeatures.
Action ended 22:19:48: UnpublishFeatures. Return value 1.
Action start 22:19:48: RemoveRegistryValues.
Action ended 22:19:48: RemoveRegistryValues. Return value 1.
Action start 22:19:48: RemoveShortcuts.
Action ended 22:19:48: RemoveShortcuts. Return value 1.
Action start 22:19:48: RemoveFiles.
Action ended 22:19:48: RemoveFiles. Return value 1.
Action start 22:19:48: RemoveFolders.
Action ended 22:19:48: RemoveFolders. Return value 1.
Action start 22:19:48: CreateFolders.
Action ended 22:19:48: CreateFolders. Return value 1.
Action start 22:19:48: InstallFiles.
Action ended 22:19:48: InstallFiles. Return value 1.
Action start 22:19:48: CreateShortcuts.
Action ended 22:19:48: CreateShortcuts. Return value 1.
Action start 22:19:48: WriteRegistryValues.
Action ended 22:19:48: WriteRegistryValues. Return value 1.
Action start 22:19:48: RegisterUser.
Action ended 22:19:48: RegisterUser. Return value 1.
Action start 22:19:48: RegisterProduct.
Action ended 22:19:48: RegisterProduct. Return value 1.
Action start 22:19:48: PublishFeatures.
Action ended 22:19:48: PublishFeatures. Return value 1.
Action start 22:19:48: PublishProduct.
Action ended 22:19:48: PublishProduct. Return value 1.
Action start 22:19:48: InstallFinalize.
MSI (s) (E4:04) [22:19:48:877]: Product: Nilesoft Shell -- Error 1925. You do not have sufficient privileges to complete this installation for all users of the machine. Log on as administrator and then retry this installation.

Error 1925. You do not have sufficient privileges to complete this installation for all users of the machine. Log on as administrator and then retry this installation.
Action ended 22:19:48: InstallFinalize. Return value 3.
Action ended 22:19:48: INSTALL. Return value 3.
Property(S): ALLUSERS = 1
Property(S): WIXUI_INSTALLDIR = INSTALLFOLDER
Property(S): INSTALLFOLDER = C:\Program Files\Nilesoft Shell\
Property(S): WixUIRMOption = UseRM
Property(S): MSIRESTARTMANAGERCONTROL = Disable
Property(S): DefaultUIFont = WixUI_Font_Normal
Property(S): ErrorDialog = ErrorDlg
Property(S): ARPURLINFOABOUT = https://nilesoft.org/
Property(S): ARPHELPLINK = https://nilesoft.org/docs/
Property(S): ARPNOMODIFY = 1
Property(S): ARPNOREPAIR = 1
Property(S): BURNMSIMODIFY = 0
Property(S): BURNMSIREPAIR = 0
Property(S): BURNMSIUNINSTALL = 1
Property(S): Manufacturer = Nilesoft
Property(S): ProductCode = {3025C475-D665-4288-99A8-3382654F7E11}
Property(S): ProductLanguage = 1033
Property(S): ProductName = Nilesoft Shell
Property(S): ProductVersion = 1.9.18
Property(S): UpgradeCode = {7BCA6512-254B-456F-8F08-AAD7508BEE00}
Property(S): SecureCustomProperties = ARPNOMODIFY;ARPNOREPAIR;BURNMSIMODIFY;BURNMSIREPAIR;BURNMSIUNINSTALL;NEWPRODUCTFOUND;UPGRADEFOUND
Property(S): IMPORTS = C:\Program Files\Nilesoft Shell\imports\
Property(S): ProgramMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\
Property(S): ProgramFiles6432Folder = C:\Program Files\
Property(S): ProgramFiles64Folder = C:\Program Files\
Property(S): TARGETDIR = D:\
Property(S): SourceDir = C:\Users\pagundu\AppData\Local\Temp\WinGet\Nilesoft.Shell.1.9.18\
Property(S): MsiLogFileLocation = C:\Users\pagundu\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Nilesoft.Shell.1.9.18-2024-12-03-22-19-48.500.log
Property(S): PackageCode = {04203AF0-96C0-4606-A225-00E6AB068ED7}
Property(S): ProductState = -1
Property(S): PackagecodeChanging = 1
Property(S): CURRENTDIRECTORY = D:\Work
Property(S): CLIENTUILEVEL = 3
Property(S): CLIENTPROCESSID = 14052
Property(S): VersionDatabase = 500
Property(S): MsiSystemRebootPending = 1
Property(S): VersionMsi = 5.00
Property(S): VersionNT = 603
Property(S): VersionNT64 = 603
Property(S): WindowsBuild = 9600
Property(S): ServicePackLevel = 0
Property(S): ServicePackLevelMinor = 0
Property(S): MsiNTProductType = 1
Property(S): WindowsFolder = C:\WINDOWS\
Property(S): WindowsVolume = C:\
Property(S): System64Folder = C:\WINDOWS\system32\
Property(S): SystemFolder = C:\WINDOWS\SysWOW64\
Property(S): RemoteAdminTS = 1
Property(S): TempFolder = C:\Users\pagundu\AppData\Local\Temp\
Property(S): ProgramFilesFolder = C:\Program Files (x86)\
Property(S): CommonFilesFolder = C:\Program Files (x86)\Common Files\
Property(S): CommonFiles64Folder = C:\Program Files\Common Files\
Property(S): AppDataFolder = C:\Users\pagundu\AppData\Roaming\
Property(S): FavoritesFolder = C:\Users\pagundu\Favorites\
Property(S): NetHoodFolder = C:\Users\pagundu\AppData\Roaming\Microsoft\Windows\Network Shortcuts\
Property(S): PersonalFolder = C:\Users\pagundu\Documents\
Property(S): PrintHoodFolder = C:\Users\pagundu\AppData\Roaming\Microsoft\Windows\Printer Shortcuts\
Property(S): RecentFolder = C:\Users\pagundu\AppData\Roaming\Microsoft\Windows\Recent\
Property(S): SendToFolder = C:\Users\pagundu\AppData\Roaming\Microsoft\Windows\SendTo\
Property(S): TemplateFolder = C:\ProgramData\Microsoft\Windows\Templates\
Property(S): CommonAppDataFolder = C:\ProgramData\
Property(S): LocalAppDataFolder = C:\Users\pagundu\AppData\Local\
Property(S): MyPicturesFolder = C:\Users\pagundu\Pictures\
Property(S): AdminToolsFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Administrative Tools\
Property(S): StartupFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup\
Property(S): StartMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\
Property(S): DesktopFolder = C:\Users\Public\Desktop\
Property(S): FontsFolder = C:\WINDOWS\Fonts\
Property(S): GPTSupport = 1
Property(S): OLEAdvtSupport = 1
Property(S): ShellAdvtSupport = 1
Property(S): MsiAMD64 = 25
Property(S): Msix64 = 25
Property(S): Intel = 25
Property(S): PhysicalMemory = 20332
Property(S): VirtualMemory = 14588
Property(S): LogonUser = pagundu
Property(S): UserSID = S-1-5-21-3916851990-2202146298-926377623-1001
Property(S): UserLanguageID = 16393
Property(S): ComputerName = THINKBOOK
Property(S): SystemLanguageID = 1033
Property(S): ScreenX = 1024
Property(S): ScreenY = 768
Property(S): CaptionHeight = 23
Property(S): BorderTop = 1
Property(S): BorderSide = 1
Property(S): TextHeight = 16
Property(S): TextInternalLeading = 3
Property(S): ColorBits = 32
Property(S): TTCSupport = 1
Property(S): Time = 22:19:48
Property(S): Date = 03-12-2024
Property(S): MsiNetAssemblySupport = 4.8.9032.0
Property(S): MsiWin32AssemblySupport = 6.3.26100.1
Property(S): RedirectedDllSupport = 2
Property(S): USERNAME = pagundu
Property(S): DATABASE = C:\WINDOWS\Installer\1dea53.msi
Property(S): OriginalDatabase = C:\Users\pagundu\AppData\Local\Temp\WinGet\Nilesoft.Shell.1.9.18\setup-x64.msi
Property(S): UILevel = 2
Property(S): ACTION = INSTALL
Property(S): ROOTDRIVE = D:\
Property(S): CostingComplete = 1
Property(S): OutOfDiskSpace = 0
Property(S): OutOfNoRbDiskSpace = 0
Property(S): PrimaryVolumeSpaceAvailable = 0
Property(S): PrimaryVolumeSpaceRequired = 0
Property(S): PrimaryVolumeSpaceRemaining = 0
Property(S): INSTALLLEVEL = 1
Property(S): SOURCEDIR = C:\Users\pagundu\AppData\Local\Temp\WinGet\Nilesoft.Shell.1.9.18\
Property(S): SourcedirProduct = {3025C475-D665-4288-99A8-3382654F7E11}
Property(S): ProductToBeRegistered = 1
MSI (s) (E4:04) [22:19:48:902]: Product: Nilesoft Shell -- Installation failed.

MSI (s) (E4:04) [22:19:48:903]: Windows Installer installed the product. Product Name: Nilesoft Shell. Product Version: 1.9.18. Product Language: 1033. Manufacturer: Nilesoft. Installation success or error status: 1603.

=== Logging stopped: 03-12-2024  22:19:48 ===
```

**Comment by Dragon1573** [comment]
> [!NOTE]
>
> Close with reason: **Already resolved**.
>
> - Resolved by #195993 

@pagunduu @stephengillie 🙏🏼

**Comment by stephengillie** [comment]
> Note
> 
> Close with reason: **Already resolved**.
> 
> * Resolved by [Fix for #195989 #195993](https://github.com/microsoft/winget-pkgs/pull/195993)
> 
> [@pagunduu](https://github.com/pagunduu) [@stephengillie](https://github.com/stephengillie) 🙏🏼
